test/nodetool/test_cleanup.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from rest_api_mock import expected_request
+import utils
+
+
+def test_cleanup(nodetool):
+    nodetool("cleanup", expected_requests=[
+        expected_request("GET", "/storage_service/keyspaces", params={"type": "non_local_strategy"},
+                         response=["ks1", "ks2"]),
+        expected_request("GET", "/storage_service/keyspaces", multiple=expected_request.ANY,
```

Comment:
C* nodetool does a strange thing: it requests all keyspaces first, then again, as it iterates over each keyspace, it requests the list again to check if the keyspace is in it. This is doubly wasteful and I didn't replicate this (mis)behaviour in Scylla-native nodetool. At the same time I didn't want to write two separate tests for the two implementations just for this. So the test accepts any number of calls to the keyspace API at the beginning to accomodate for this. Note that I already added this capability in a previous PR. I needed to add ANY here, because the first request happens with a param, which the subsequent requests (that only C* sends) don't have.

Reply:
Ok, then.
Another thing you could have done was to call this a "cassandra_nodetool_bug" - the expected behavior has just one request, which happens on Scylla. Because Cassandra nodetool does two requests, it will fail. Of course, in such a case we should explain why we believe this is a bug in Nodetool and not in our new implementation.
But it's not important now. You test is good too, with just one minor downside: It won't catch a regression that will cause our nodetool to start sending the request twice. But I don't supposed such a weird regression can happen, so it's ok.

Reply:
Yes. Furthermore, if I allowed this test to fail for the legacy nodetool, it could cover regressions in it, unless I write each test which encounter this twice. Which is something I want to avoid.